sdk/schemaregistry/schema-registry/test/public/utils/recordedClient.ts

```diff
@@ -1,50 +1,30 @@
 // Copyright (c) Microsoft Corporation.
 // Licensed under the MIT license.
 
-import { Context } from "mocha";
-
-import { env, Recorder, record, RecorderEnvironmentSetup } from "@azure-tools/test-recorder";
-import { ClientSecretCredential } from "@azure/identity";
-
+import { createTestCredential } from "@azure-tools/test-credential";
 import { SchemaRegistryClient } from "../../../src";
+import {
+  Recorder,
+  RecorderStartOptions,
+  getEnvironmentVariable,
+} from "@azure-tools/test-recorder-new";
 
-export interface RecordedClient {
-  client: SchemaRegistryClient;
-  recorder: Recorder;
-}
-
-const replaceableVariables: { [k: string]: string } = {
-  AZURE_CLIENT_ID: "azure_client_id",
-  AZURE_CLIENT_SECRET: "azure_client_secret",
-  AZURE_TENANT_ID: "azuretenantid",
-  SCHEMA_REGISTRY_ENDPOINT: "https://endpoint",
-  SCHEMA_REGISTRY_GROUP: "group-1",
-};
-
-const environmentSetup: RecorderEnvironmentSetup = {
-  replaceableVariables,
-  customizationsOnRecordings: [
-    (recording: string): string =>
-      recording.replace(/"access_token"\s?:\s?"[^"]*"/g, `"access_token":"access_token"`),
-    // If we put ENDPOINT in replaceableVariables above, it will not capture
-    // the endpoint string used with nock, which will be expanded to
-    // https://<endpoint>:443/ and therefore will not match, so we have to do
-    // this instead.
-    (recording: string): string => {
-      const replaced = recording.replace("endpoint:443", "endpoint");
-      return replaced;
-    },
-  ],
-  queryParametersToSkip: [],
+export const recorderOptions: RecorderStartOptions = {
+  envSetupForPlayback: {
+    AZURE_CLIENT_ID: "azure_client_id",
+    AZURE_CLIENT_SECRET: "azure_client_secret",
+    AZURE_TENANT_ID: "azuretenantid",
+    SCHEMA_REGISTRY_ENDPOINT: "https://endpoint",
+    SCHEMA_REGISTRY_GROUP: "group-1",
+  },
 };
 
-export function createRecordedClient(context: Context): RecordedClient {
-  const recorder = record(context, environmentSetup);
-  const credential = new ClientSecretCredential(
-    env.AZURE_TENANT_ID,
-    env.AZURE_CLIENT_ID,
-    env.AZURE_CLIENT_SECRET
+export function createRecordedClient(recorder: Recorder): SchemaRegistryClient {
+  const credential = createTestCredential();
+  const client = new SchemaRegistryClient(
+    getEnvironmentVariable("SCHEMA_REGISTRY_ENDPOINT"),
+    credential
   );
-  const client = new SchemaRegistryClient(env.SCHEMA_REGISTRY_ENDPOINT, credential);
-  return { client, recorder };
+  recorder.configureClient(client["client"]);
```

Comment:
Agh. I hate to say this, but I am really opposed to this way of accessing private fields. I would rather cast the object to an `any` than use this element-indexer hack.

Can we instead make whatever modification is necessary to be able to expose the pipeline? I don't think we should rush into this if the UX for accessing `pipeline` is not set.

Reply:
@witemple-msft this is meant as a stop gap until the architects agree on how to expose the pipeline.

I am curious though, is your preference to cast to any just philosophical or is there something else? btw it is documented in the [official docs](https://www.typescriptlang.org/docs/handbook/2/classes.html#caveats) so I like to think of it as just a limitation of the typechecker.

Reply:
I don't think it's a limitation but an intended feature of the type checker that it ignores privacy modifiers when using element access, but IMO it's not something we want to rely on.  It's a totally bizarre feature that `x.y` and `x["y"]` are totally equivalent in almost every way in TypeScript _except that it ignores privacy modifiers_.

If we start migrating packages now and using this hack, will we go back and migrate all packages _again_ when we have a set way of exposing the pipeline publicly? It seems like we should try to get the pipeline API right first instead of hacking into private fields that won't be consistent across packages.

Maybe I'm being too conservative here, but I feel that we ought to have some standard way to access the pipeline that is uniform across _all_ corev2 packages so that we can just _always_ call `recorder.configureClient(client)`.

Reply:
> It seems like we should try to get the pipeline API right first instead of hacking into private fields that won't be consistent across packages.

Well yeah that would be ideal, but this design discussion has been around for a few months now with no conclusion yet. I would rather use the new recorder now and update this line later tbh since I already have so many issues with the old recorder.

Reply:
I added my thoughts to the issue. Good discussion, friends!

Reply:
https://github.com/Azure/azure-sdk-for-js/issues/16876